components/style/values/specified/font.rs

```diff
@@ -988,6 +989,113 @@ impl From<FontSynthesis> for u8 {
     }
 }
 
+#[derive(Clone, Debug, Eq, MallocSizeOf, PartialEq)]
+/// Allows authors to explicitly specify the language system of the font,
+/// overriding the language system implied by the content language
+pub enum FontLanguageOverride {
+    /// When rendering with OpenType fonts,
+    /// the content language of the element is
+    /// used to infer the OpenType language system
+    Normal,
+    /// Single three-letter case-sensitive OpenType language system tag,
+    /// specifies the OpenType language system to be used instead of
+    /// the language system implied by the language of the element
+    Override(String),
+    /// Use system font
+    System(SystemFont)
+}
+
+impl FontLanguageOverride {
+    #[inline]
+    /// Get default value with `normal`
+    pub fn normal() -> FontLanguageOverride {
+        FontLanguageOverride::Normal
+    }
+
+    /// Get `font-language-override` with `system font`
+    pub fn system_font(f: SystemFont) -> Self {
+        FontLanguageOverride::System(f)
+    }
+
+    /// Get system font
+    pub fn get_system(&self) -> Option<SystemFont> {
+        if let FontLanguageOverride::System(s) = *self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+}
+
+impl ToCss for FontLanguageOverride {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        match *self {
+            FontLanguageOverride::Normal => dest.write_str("normal"),
+            FontLanguageOverride::Override(ref lang) => lang.to_css(dest),
+            FontLanguageOverride::System(sys) => sys.to_css(dest),
+        }
+    }
```

Comment:
After checking #19186 , I think I can remove this `impl` and `derive` `ToCss` instead?

Reply:
Indeed this can just be derived, mind doing that? Thanks! :)

Reply:
Sure, I'll change it to use `derive` in the late today :)